langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Matches.java

```diff
@@ -53,7 +65,20 @@ public static BArray matchGroupsAt(BRegexpValue regExp, BString str, int startIn
     }
 
     public static boolean isFullMatch(BRegexpValue regExp, BString str) {
-        Matcher matcher = RegexUtil.getMatcher(regExp, str);
+        if (regExp.stringValue(null).equals("")) {
+            return false;
+        }
+
+        Matcher matcher = getMatcher(regExp, str);
         return matcher.matches();
     }
+
+    private static Matcher getMatcher(BRegexpValue regExp, BString str) {
+        try {
+            return RegexUtil.getMatcher(regExp, str);
+        } catch (PatternSyntaxException e) {
+            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.REG_EXP_PARSING_ERROR,
```

Comment:
Can't we handle this at the util level where we actually create the Pattern?

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/d2e825dc93401a2bcd7bac4ee797b9304347c3ee.